### PR TITLE
modified json schema and put into a separate file

### PR DESCRIPTION
**Pull request by Hemant27031999**
## What does it fixes?
This PR is solving issue #720.

## Additional fixes.
I have also extracted the json schema from `json.py` file and have put it in a separate file.
Modifying the json schema also passed the two test cases inside `test_json.py` which were initially marked as xfail.

### CODE REVIEW

**Comment by Hemant27031999** [comment]
@matthiaskoenig @draeger @Midnighter @cdiener. Please review it.

File: cobra/io/json_schemas/json_schema_v1.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
+
+json_schema_V1 = {
```
Comment (Midnighter) [inline cobra/io/json_schemas/json_schema_v1.py]: Please make this just an object inside of a JSON file.

File: cobra/test/test_io/test_json.py [hunk 2]
```diff
@@ -10,17 +10,17 @@
 import pytest
 
 import cobra.io as cio
+from cobra.io.json_schemas.json_schema_v1 import json_schema_V1
```
Comment (Midnighter) [inline cobra/test/test_io/test_json.py]: This will have to be changed to loading the schema directly.

File: cobra/test/test_io/test_json.py [hunk 3]
```diff
@@ -29,13 +29,12 @@ def test_load_json_model(data_directory, mini_model):
...
 
-@pytest.mark.xfail(reason="schema outdated")
 def test_save_json_model(tmpdir, mini_model):
     """Test the writing of JSON model."""
     jsonschema = pytest.importorskip("jsonschema")
     output_file = tmpdir.join("mini.json")
     cio.save_json_model(mini_model, output_file.strpath, pretty=True)
     # validate against JSONSchema
-    with open(output_file, "r") as infile:
+    with open(output_file.strpath, "r") as infile:
```
Comment (Midnighter) [inline cobra/test/test_io/test_json.py]: The following seems more natural to me.

```suggestion
    with open(str(output_file), "r") as infile:
```

**Comment by Midnighter** [comment]
You will also have to modify the `setup.py` to:

```py
        package_data={
             '': [
                 'cobra/io/*.json',
                 'test/data/*',
                 'mlab/matlab_scripts/*m'
             ]
        },

```

**Comment by matthiaskoenig** [comment]
@Hemant27031999 Can you implement the proposed changes so the pull request can be merged?

**Comment by Hemant27031999** [comment]
Nice to hear from you, @Midnighter. 
I will send the updated code in the next few hours.

**Comment by Hemant27031999** [comment]
I have made the suggested changes @Midnighter @matthiaskoenig .

File: cobra/test/conftest.py [hunk 4]
```diff
@@ -74,6 +74,14 @@ def salmonella(medium_model):
     return medium_model.copy()
 
 
+@pytest.fixture(scope="session")
```
Comment (Midnighter) [inline cobra/test/conftest.py:77]: 1. Please move this fixture into the `cobra/test/test_io/test_json.py` module.
2. I outlined how to load the schema with importlib resources. Did you try that?

Comment (Hemant27031999) [reply cobra/test/conftest.py:77]: I did try using 'importlib_resources' also. The tests were passing but the tox commands were failing with the error: ImportError: No module named importlib_resources

Comment (Midnighter) [reply cobra/test/conftest.py:77]: Okay, I can take it from here by adding another commit.

**Pull request merged**